### PR TITLE
PR for #213: Automatic redirect to login

### PR DESCRIPTION
**Pull request by jxiao21**
Made this PR so that we can continue on code changes from #401

### CODE REVIEW

**Comment by jaydonkrooss** [comment]
I'll work on reviewing. as Matt suggested, the formatting on base.html is a little off from the original version, so maybe try to restore the spacing that was on the <head> tag

File: src/officehours_ui/templates/base.html [hunk 1]
```diff
@@ -3,112 +3,140 @@
...
+ <link rel="stylesheet" type="text/css" href="{% static 'css/main.css' %}">
+ <link rel="stylesheet" mo type="text/css" href="//fonts.googleapis.com/css?family=Roboto:400"/>
+ <link rel="stylesheet" type="text/css" href="//fonts.googleapis.com/css?family=Roboto:400i"/>
+ <link rel="stylesheet" type="text/css" href="//fonts.googleapis.com/css?family=Roboto:700"/>
+ <link rel="stylesheet" type="text/css" href="//fonts.googleapis.com/css?family=Roboto:700i"/>
+ {% block styles %}{% endblock %}
+ <title>Remote Office Hours Queue - {% block title %}{% endblock %}</title>
+ {{ spa_globals|json_script:"spa_globals" }}
+ {% endblock %}
+ </head>
```
Comment (jaydonkrooss) [inline src/officehours_ui/templates/base.html:22]: these are the formatting changes you made, maybe try copying and pasting from the old version and see if the diff changes

Comment (jxiao21) [reply src/officehours_ui/templates/base.html]: Ok I think it should be good now let me know if I missed something!

Comment (jxiao21) [reply src/officehours_ui/templates/base.html]: I don't know if it's exactly how it was before but at least it hopefully looks more intuitive than whatever it was when I originally pushed it

**Comment by jonespm** [comment]
Note: When this is merged please include

Co-authored-by: tayheath (Taylor Heath) <tayheath@umich.edu>

File: src/officehours_ui/templates/base.html [hunk 2]
```diff
@@ -21,94 +21,121 @@
...
-  <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.2.3/dist/js/bootstrap.min.js" integrity="sha384-cuYeSxntonz0PPNlHhBs68uyIAVpIIOZZ5JqeqvYYIcEL727kskC66kF92t6Xl2V" crossorigin="anonymous"></script>
-  {% block scripts %}{% endblock %}
 
-  {% get_flatpages '/dataalert/' as flatpages %}
-  {% if flatpages.first and flatpages.first.content %}
-    <footer class="sticky-footer">
-      <div class="alert alert-info alert-footer" role="alert">
-        <div class="alert-footer-text">
-          {{flatpages.first.content|safe}}
+    <script src="https://code.jquery.com/jquery-3.3.1.slim.min.js" integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo" crossorigin="anonymous"></script>
```
Comment (jaydonkrooss) [inline src/officehours_ui/templates/base.html:102]: @jonespm has maybe mentioned this before, but I'm not seeing anywhere in the code that actually makes use of the JQuery import. If everything behaves as intended without this, I suggest removing it.

File: src/officehours_ui/templates/base.html [hunk 3]
```diff
@@ -21,94 +21,121 @@
...
+      if (logoutButton !== null) {
+        logoutButton.addEventListener('click', () => {
+          window.localStorage.setItem("loggedIn", "false");
+        });
+      }
+      window.addEventListener('storage', (event) => {
+        //  if (document.cookie.indexOf('JSESSIONID') === -1) {
+        //  if (document.cookie.indexOf('sessionid') === -1) {
+        //    window.location.href = "/accounts/login/";
+        //  }
```
Comment (jaydonkrooss) [inline src/officehours_ui/templates/base.html:117]: Remove unused code in comments

**Comment by jaydonkrooss** [comment]
Logic is looking like it's working, I tested logging out on different browsers and the redirect behaved as expected

File: src/officehours_ui/templates/base.html (hunk 2 above)
Comment (jxiao21) [reply src/officehours_ui/templates/base.html]: Appears to work without JQuery so I removed it

File: src/officehours_ui/templates/base.html (hunk 3 above)
Comment (jxiao21) [reply src/officehours_ui/templates/base.html]: Done and pushed

**Review by jaydonkrooss — APPROVED** [review]

**Comment by jxiao21** [comment]
Test plan:
1. Open ROHQ on two different tabs (ex. 2 different Chrome tabs; I don't know if it's possible to fix this issue on, for example, one Chrome tab and one Safari tab).
2. Make sure you're logged in on both tabs.
3. Log out on one tab, then check the other tab. It should be logged out now as well.

**Pull request merged**